### ISSUE TITLE
[Helm] argo-cd helm chart lock file cannot be updated

### ISSUE THREAD

**Issue by TilmannBach** [renovatebot/renovate#13858]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.56.0

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab 14.7.0

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Renovate Bot started to report an ` Artifact update problem` since argo-cd helm chart version 3.32.3

this is my gitlab merge request log:

```
sa_faro (Busy) 🤖 @sa_faro changed title from chore(deps): update helm release argo-cd to v3.31.1 to chore(deps): update helm release argo-cd to v3.32.0 21 hours ago
sa_faro (Busy) 🤖 @sa_faro changed the description 21 hours ago · 
sa_faro
sa_faro (Busy) 🤖 @sa_faro · 21 hours ago
Author
Maintainer

⚠ Artifact update problem
Renovate failed to update an artifact related to this branch. You probably do not want to merge this MR as-is.

♻ Renovate will retry this branch, including artifacts, only when one of the following happens:

any of the package files in this branch needs updating, or
the branch becomes conflicted, or
you click the rebase/retry checkbox if found above, or
you rename this MR's title to start with "rebase!" to trigger it manually
The artifact failure details are included below:

File name: charts/argo-cd/Chart.lock
Command failed: helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd
Error: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory
sa_faro (Busy) 🤖 @sa_faro added 1 commit 2 hours ago
60c8622e - chore(deps): update helm release argo-cd to v3.32.1
```

i also tried to tick the retry button and renovate retries but stuck with the same error (see error log).

this worked before yesterday afternoon (and with argo cd helm chart versions < 3.32.0

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Value mismatch (repository=faro/deployment/deployment-local, packageFile=charts/argo-cd/Chart.yaml, branch=renovate/argo-cd-3.x)
       "manager": "helmv3",
       "expectedValue": "3.32.0",
       "foundValue": "3.31.1"
DEBUG: Rebasing after outdated branch dep found (repository=faro/deployment/deployment-local, packageFile=charts/argo-cd/Chart.yaml, branch=renovate/argo-cd-3.x)
       "depName": "argo-cd"
DEBUG: manager.getUpdatedPackageFiles() reuseExistinbranch=false (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
DEBUG: Starting search at index 90 (repository=faro/deployment/deployment-local, packageFile=charts/argo-cd/Chart.yaml, branch=renovate/argo-cd-3.x)
       "depName": "argo-cd"
DEBUG: Found match at index 90 (repository=faro/deployment/deployment-local, packageFile=charts/argo-cd/Chart.yaml, branch=renovate/argo-cd-3.x)
       "depName": "argo-cd"
DEBUG: Contents updated (repository=faro/deployment/deployment-local, packageFile=charts/argo-cd/Chart.yaml, branch=renovate/argo-cd-3.x)
       "depName": "argo-cd"
DEBUG: helmv3.updateArtifacts(charts/argo-cd/Chart.yaml) (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
DEBUG: Updating charts/argo-cd/Chart.lock (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
DEBUG: Executing command (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "command": "helm repo add stable --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories https://charts.helm.sh/stable"
DEBUG: exec completed (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "cmd": "helm repo add stable --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories https://charts.helm.sh/stable",
       "durationMs": 2651,
       "stdout": "\"stable\" has been added to your repositories\n",
       "stderr": ""
DEBUG: Executing command (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "command": "helm repo add argo-cd --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories --username sa_faro2 --password **redacted** https://artifactory.mycompany.exmaple.com/artifactory/api/helm/argo-helm"
DEBUG: exec completed (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "cmd": "helm repo add argo-cd --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories --username sa_faro2 --password **redacted** https://artifactory.mycompany.exmaple.com/artifactory/api/helm/argo-helm",
       "durationMs": 385,
       "stdout": "\"argo-cd\" has been added to your repositories\n",
       "stderr": ""
DEBUG: Executing command (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "command": "helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd"
DEBUG: rawExec err (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd",
         "stdout": "Getting updates for unmanaged Helm repositories...\n...Successfully got an update from the \"https://argoproj.github.io/argo-helm\" chart repository\nHang tight while we grab the latest from your chart repositories...\n...Successfully got an update from the \"argo-cd\" chart repository\n...Successfully got an update from the \"stable\" chart repository\nUpdate Complete. ⎈Happy Helming!⎈\n",
         "stderr": "Error: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory\n",
         "message": "Command failed: helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd\nError: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory\n",
         "stack": "Error: Command failed: helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd\nError: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory\n\n    at ChildProcess.exithandler (child_process.js:383:12)\n    at ChildProcess.emit (events.js:400:28)\n    at ChildProcess.emit (domain.js:475:12)\n    at maybeClose (internal/child_process.js:1058:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)"
       }
DEBUG: Failed to update Helm lock file (repository=faro/deployment/deployment-local, branch=renovate/argo-cd-3.x)
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd",
         "stdout": "Getting updates for unmanaged Helm repositories...\n...Successfully got an update from the \"https://argoproj.github.io/argo-helm\" chart repository\nHang tight while we grab the latest from your chart repositories...\n...Successfully got an update from the \"argo-cd\" chart repository\n...Successfully got an update from the \"stable\" chart repository\nUpdate Complete. ⎈Happy Helming!⎈\n",
         "stderr": "Error: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory\n",
         "message": "Command failed: helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd\nError: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory\n",
         "stack": "Error: Command failed: helm dependency update --registry-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/registry.json --repository-config /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories charts/argo-cd\nError: no cached repository for helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf found. (try 'helm repo update'): open /builds/faro/renovate-bot/renovate/cache/__renovate-private-cache/repositories/helm-manager-2f78de71dc5ee04843b948cb36875a06dc6a375e69cdb87729a726444558cccf-index.yaml: no such file or directory\n\n    at ChildProcess.exithandler (child_process.js:383:12)\n    at ChildProcess.emit (events.js:400:28)\n    at ChildProcess.emit (domain.js:475:12)\n    at maybeClose (internal/child_process.js:1058:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)"
       }

```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Can you check if this was fixed by #13848 in v31.58.1 ?

**Comment by TilmannBach** [comment]
hi i tested this with 31.58.5 today and the error has gone, so we can close this issue, but i have a another one now 😕. i will open a new issue for that.
